Fixed 'Orbot appears not to be running' (#1792)

vanitasvitae opened the pull request:
Hi!
I tried my best to fix the faulty "Orbot appears not to be running" message on Android 6 by ditching the old OrbotHelper.isOrbotRunning() method. I implemented a BroadcastReceiver that listens for Orbot Status intents and acts as a singleton with getters for the received states.

Im unsure, whether this is secure and proper implemented or not, so please take a close look at my changes :)

dschuermann commented:
Looks good.
Does this work properly in your tests?

dschuermann commented on OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java:
```diff
@@ -0,0 +1,78 @@
+import org.sufficientlysecure.keychain.util.Log;
+
+/**
+ * Created by vanitas on 11.05.16.
```
remove the "created by" line, instead add a license header like in other files with your name

dschuermann commented on OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java:
```diff
@@ -0,0 +1,78 @@
+ */
+public class OrbotStatusReceiver extends BroadcastReceiver {
+
+    private static final String TAG = "OrbStatRec";
```
use Constants.TAG instead

dschuermann commented on OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    public static OrbotStatusReceiver getInstance() {
+        if(instance == null) instance = new OrbotStatusReceiver();
```
if else always with {}

dschuermann commented on OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/TorServiceUtils.java:
```diff
@@ -1,156 +0,0 @@
-/* This is the license for Orlib, a free software project to
```
great to get rid of this file :) I don't like it

vanitasvitae commented:
I changed the points you mentioned. Im not sure what you meant with the authorship so I removed it completely.

I tested this manually on my device by deactivating/starting Orbot. Are there any test cases which I can use or should I write my own?

dschuermann commented:
Great work!

dschuermann commented:
Regarding tests: I don't think this can be tested in an automated manner.

vanitasvitae commented:
Is seems like OpenKeyChain is checking Orbots status nearly every second. I'm going to take a look at this since it may drain battery.
